### ISSUE TITLE
Trailing white spaces can make filters not work

### ISSUE THREAD

**Issue by Takiro** [mastodon/mastodon#23789]
### Steps to reproduce the problem

1. Create a new filter in Settings → Filters  and deliberately add a trailing white space
2. Save the filter
3. Check your Timeline and if the filter is hiding posts containing the filtered words

### Expected behaviour

All posts with the words defined in the filter should be hidden.

### Actual behaviour

Some posts will still appear. 

### Detailed description

This looks like its obvious since of course the filter expression is taken literally. The filter will sometimes work, (when the filter expression is in a middle or beginning of a sentence) and sometimes not (for example when the word you actually want to block is followed by punctuation). 

However a trailing white space is easy to overlook and I already had it happen to me several times so that I thought the filter wasn't working at all. That the culprit is a trailing (or leading) white space is not obvious to the user and the resulting behavior confusing. 

I suggest to either make white space characters visible or trimming the filter expression when saving the filter to prevent this from happening. 

### Specifications

Mastodon v4.1.0

**Comment by ClearlyClaire** [comment]
Good catch! I'd be in favor of just stripping surrounding spaces. Need to think a bit more about whether this would break any use case, but I think it's pretty unlikely.

**Comment by roycewilliams** [comment]
See [comment on PR #23883](https://github.com/mastodon/mastodon/pull/23883#issuecomment-1666569965) - I'd like to propose keeping leading/trailing whitespace, but just making it visible instead.

**Comment by ClearlyClaire** [comment]
That would be ideal, but I have no idea how to make those whitespaces more visible.
